### DISCUSSION TITLE
UniGetUI 3.1.0 beta 1

### DISCUSSION THREAD

**Discussion by marticliment** [Devolutions/UniGetUI#2372] [Announcements]
![](https://img.shields.io/github/downloads/marticliment/wingetui/3.1.0-beta1/total?style=for-the-badge&color=red)

<b>IMPORTANT:
This is the second beta release to implement the UniGetUI rebranding.
Instaling this release will automatically migrate settings and other data to new folders (WingetUI -> UniGetUI), so rolling back to older versions of WingetUI may cause settings to be reset.
</b>

NOTE: This release attempts to fix an unexpected issue with the WinGet package manager interop. Assuming no more issues are found after this release, UniGetUI 3.1.0 stable will be published.
## Changes 
 - This release adds a built-in dependency manager, that checks and prompts the user to install missing dependencies for UniGetUI (let's say, for example, Microsoft.WinGet.Client or Scoop-Search).
 - Other minor issues have been fixed on this release.

SHA256: `CAFE3438B5852C32A6E18CE52C3B2848F968530E0A13ABA73240624E29766FD5`

<hr /><em>This discussion was created from the release <a href='https://github.com/marticliment/WingetUI/releases/tag/3.1.0-beta1'>UniGetUI 3.1.0 beta 1</a>.</em>

**Comment by 1aTa** [comment]
I'm getting a missing dependency error with this version:

"UniGetUI requires .NET Tools Outdated operate, but it was not found on your system" and a button to "Install .NET Tools Outdated" which doesn't seem to do anything.

I already have the latest .NET desktop runtime v8.0.6 installed in the correct location.

**Reply by marticliment** [reply]
I am aware of this issue, and I have fix it in code. I will publish another beta with this issue patched, as well as with the ability to ignore missing dependencies

**Comment by Denis-Maes** [comment]
Bonjour,

FeedBack :
j'ai formaté mon pc, quelques jours après, j'installe la version UniGetUI 3.1.0b1.
Je dois installer "Microsoft.WinGet.Client" mais en vain. J'ai aussi essayé manuellement via le terminal admin, idem.
Je ne sais pas si c'est grave pour le bon fonctionnement du logiciel. Je tenais à vous l'informer.

PS :
j'adore ce programme et je vous félicite.

**Reply by marticliment** [reply]
Hello,

Yes, UniGetUI requires from the Microsoft.WinGet.Client to work. Can you please share your UniGetUI logs?

**Comment by Denis-Maes** [comment]
Oui, bien sûr.
Dite moi où se trouvent les journaux UniGetUI.

**Reply by marticliment** [reply]
![image](https://github.com/marticliment/WingetUI/assets/53119851/2c5d03d6-f23d-4911-b09d-b6eb571b19e3)

**Comment by Denis-Maes** [comment]
[02/07/2024 12:45:56] 
                         __  __      _ ______     __  __  ______
                        / / / /___  (_) ____/__  / /_/ / / /  _/
                       / / / / __ \/ / / __/ _ \/ __/ / / // /  
                      / /_/ / / / / / /_/ /  __/ /_/ /_/ // /   
                      \____/_/ /_/_/\____/\___/\__/\____/___/   
                          Welcome to UniGetUI Version 3.1.0-beta1
[02/07/2024 12:45:56]
[02/07/2024 12:45:56] Version Code:  3,0993
[02/07/2024 12:45:56] Using bundled GSudo at C:\Program Files\UniGetUI\Assets\Utilities\gsudo.exe
[02/07/2024 12:45:56] Loaded language locale: fr
[02/07/2024 12:45:56] Randomly-generated background API auth token for the current session: y7fua87nfw11z6wzk6l9sja9nmttyfdq8rbhksphczaaj77ov6n9v0hq91ku8dvn
[02/07/2024 12:45:56] Api running on http://localhost:7058
[02/07/2024 12:45:56] Lang files were updated successfully from GitHub
[02/07/2024 12:45:56] UniGetUI is up to date
[02/07/2024 12:45:56] Downloaded new icons and screenshots successfully!
[02/07/2024 12:45:56] Command npm was not found on the system
[02/07/2024 12:45:56] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Npm
                      █ Enabled: True
                      █ Found: False
                      █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
                      █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[02/07/2024 12:45:56] Command dotnet.exe was not found on the system
[02/07/2024 12:45:56] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: .NET Tool
                      █ Enabled: True
                      █ Found: False
                      █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
                      █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[02/07/2024 12:45:57] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Pip
                      █ Enabled: True
                      █ Found: True
                      █ Fancye exe name: pip
                      █ Executable path: C:\Users\denis\AppData\Local\Microsoft\WindowsApps\python.exe
                      █ Call arguments:  -m pip
                      █ Version: 
                      █   
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[02/07/2024 12:45:58] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Winget
                      █ Enabled: True
                      █ Found: True
                      █ Fancye exe name: winget.exe
                      █ Executable path: C:\Users\denis\AppData\Local\Microsoft\WindowsApps\winget.exe
                      █ Call arguments: 
                      █ Version: 
                      █   Naive WinGet CLI Version: v1.8.1791
                      █   Microsoft.WinGet.Client PSModule version: 
                      █   Using Native WinGet helper (COM Api)
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[02/07/2024 12:45:58] Command scoop was not found on the system
[02/07/2024 12:45:58] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Scoop
                      █ Enabled: True
                      █ Found: False
                      █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
                      █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[02/07/2024 12:45:59] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Chocolatey
                      █ Enabled: True
                      █ Found: True
                      █ Fancye exe name: choco.exe
                      █ Executable path: C:\Users\denis\AppData\Local\UniGetUI\Chocolatey\choco.exe
                      █ Call arguments: 
                      █ Version: 
                      █   2.2.2
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[02/07/2024 12:46:00] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: PowerShell
                      █ Enabled: True
                      █ Found: True
                      █ Fancye exe name: powershell.exe
                      █ Executable path: C:\WINDOWS\system32\windowspowershell\v1.0\powershell.exe
                      █ Call arguments:  -NoProfile -Command
                      █ Version: 
                      █   Name                           Value                                                                                   
                      █   ----                           -----                                                                                   
                      █   PSVersion                      5.1.22621.3810                                                                          
                      █   PSEdition                      Desktop                                                                                 
                      █   PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
                      █   BuildVersion                   10.0.22621.3810                                                                         
                      █   CLRVersion                     4.0.30319.42000                                                                         
                      █   WSManStackVersion              3.0                                                                                     
                      █   PSRemotingProtocolVersion      2.3                                                                                     
                      █   SerializationVersion           1.1.0.1
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[02/07/2024 12:46:00] Found 0 available updates from Pip
[02/07/2024 12:46:00] Found 0 installed packages from Pip
[02/07/2024 12:46:02] Dependency Microsoft.WinGet.Client was not found for manager Winget, installing...
[02/07/2024 12:46:03] Found 0 installed packages from Chocolatey
[02/07/2024 12:46:03] Found 0 available updates from Chocolatey
[02/07/2024 12:46:03] Found 0 installed packages from Winget
[02/07/2024 12:46:03] Found 0 installed packages from PowerShell
[02/07/2024 12:46:03] Found 0 available updates from PowerShell
[02/07/2024 12:46:03] Found 0 available updates from Winget
[02/07/2024 12:51:11] LoadComponentsAsync finished executing. All managers loaded. Proceeding to interface.

**Reply by marticliment** [reply]
Please try to run UniGetUI as admin and, when prompted, install the missing Microsoft.WinGet.Client dependency

**Comment by Denis-Maes** [comment]
Ça fonctionne, et maintenant, j'ai des mises à jour.
Je vais noter cette manipulation.

Merci
